// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package wireless.android.adservices.mdd.adtech_enrollment.chrome;

enum PrivacySandboxAttestationsGatedAPIProto {
  UNKNOWN = 0;
  TOPICS = 1;
  PROTECTED_AUDIENCE = 2;
  PRIVATE_AGGREGATION = 3;
  ATTRIBUTION_REPORTING = 4;
  SHARED_STORAGE = 5;
  FENCED_STORAGE_READ = 6;
}

message PrivacySandboxAttestationsProto {
  message PrivacySandboxAttestedAPIsProto {
    repeated PrivacySandboxAttestationsGatedAPIProto attested_apis = 1;
  }

  // A definition of what "all APIs" means for the current version.
  repeated PrivacySandboxAttestationsGatedAPIProto all_apis = 1;

  // A list of sites attested for all APIs.
  repeated string sites_attested_for_all_apis = 2;

  // A map from sites to lists of attested APIs. (Should only be used for sites
  // that are not attested for all APIs.)
  map<string, PrivacySandboxAttestedAPIsProto> site_attestations = 3;
}
